_lessons/python-scatter/06-profiling_with_MAP.md

```diff
@@ -1,77 +1,79 @@
+ * *tracing*: records the activity of instrumented parts of the code in time. This is useful to determine what each thread and process is doing at a particaular point in time. Tracing can reveal load balancing issues or the communication patterns of an MPI program. 
 
-Due to possible overhead (especially for tracing experiments) the program could run significantly slower than normal. Therefore it is generally advisable to first run a sampling experiment and choose a small representative test case to profile.
+MAP can be used both for sampling and tracing.
 
-## Profiling test case
+## Choosing your test case
 
-The profiling case should be:
-* representative, need to cover all features used during production
-* short in run time for a fast turnaround, but long enough to be representative of the actual calculations you are going to run
-* scaling effects can be investigated by comparing profiles small and big jobs
+Choose your test case carefully, making sure:
+* it runs in a short time. Due to a possible large overhead from MAP (especially for tracing experiments), the program could run significantly slower than normal.
+* ensure the test case is representative of a production run
 
-**Note:** keep in mind, that with shortened computation time, the initialisation and finalisation becomes more dominant.
+**Note:** keep in mind, that with shortened computation time, the initialisation and finalisation steps may become dominant.
 
 
-## ARM MAP profiler
+## MAP profiler
 
-On NeSI systems the [ARM MAP](https://www.arm.com/products/development-tools/server-and-hpc/forge/map) profiler is provided as part of the `forge` module (along with the parallel debugger, DDT).
+On NeSI systems the [ARM MAP](https://www.arm.com/products/development-tools/server-and-hpc/forge/map) profiler is provided as part of the *forge* module (along with the parallel debugger DDT).
 
-MAP is a commercial product, which can profile parallel, multi-threaded and single-threaded C/C++, Fortran and F90, as well as Python codes. It can be used without code modification.
-MAP can be launched with a GUI and without. The GUI allows the user to navigate through the code and enables them to focus on specific source lines. The "Express Launch", without GUI enables easy usage of existing queue scripts and workflows.
+MAP is a commercial product, which can profile parallel, multi-threaded and single-threaded C/C++, Fortran, as well as Python code. It can be used without code modification.
+MAP can be launched with a graphical user interface and without. The graphical user interface allows the user to navigate through the code and focus on specific source lines. The "Express Launch", without graphical user interface, makes it ieasy to submit job scripts and workflows.
```

Comment:
I would first "graphical user interface (GUI)" and then stay with GUI. 
Furthermore, there is a spelling mistake with ieasy. But actually, we want to prevent using such terms. I would say: "The 'Express Launch', without GUI, allows to maintain workflow, since it just requires a modification of the execution line (*srun* command)."

Reply:
OK thanks.